lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -827,114 +912,37 @@ def remove_toolitem(self, name):
                     self._groups[group].remove(toolitem)
         del self._toolitems[name]
 
+    def _update_flow(self):
+        self.set_property("orientation", _flow[not self._flow])
+        self._toolarea.set_property('orientation', _flow[self._flow])
+        for item in self._toolarea:
+            if isinstance(item, Gtk.Separator):
+                item.set_property("orientation", _flow[not self._flow])
+            else:
+                item.set_property("orientation", _flow[self._flow])
+
     def _add_separator(self):
         sep = Gtk.Separator()
-        sep.set_property("orientation", Gtk.Orientation.VERTICAL)
+        sep.set_property("orientation", _flow[not self._flow])
         self._toolarea.pack_start(sep, False, True, 0)
         sep.show_all()
 
 
 class StatusbarGTK3(StatusbarBase, Gtk.Statusbar):
     def __init__(self, *args, **kwargs):
-        StatusbarBase.__init__(self, *args, **kwargs)
-        Gtk.Statusbar.__init__(self)
+        super(StatusbarGTK3, self).__init__(*args, **kwargs)
         self._context = self.get_context_id('message')
 
     def set_message(self, s):
         self.pop(self._context)
         self.push(self._context, s)
 
 
-class SaveFigureGTK3(backend_tools.SaveFigureBase):
```

Comment:
There is a new generic Tool that does exactly that without having to implement it for each backend.